logger/dapr_logger.go

```diff
@@ -112,19 +123,38 @@ func (l *daprLogger) SetOutput(dst io.Writer) {
 // WithLogType specify the log_type field in log. Default value is LogTypeLog.
 func (l *daprLogger) WithLogType(logType string) Logger {
 	return &daprLogger{
-		name:   l.name,
-		logger: l.logger.WithField(logFieldType, logType),
+		name:         l.name,
+		traceEnabled: defaultTraceEnabled,
+		logger:       l.logger.WithField(logFieldType, logType),
 	}
 }
 
 // WithFields returns a logger with the added structured fields.
 func (l *daprLogger) WithFields(fields map[string]any) Logger {
 	return &daprLogger{
-		name:   l.name,
-		logger: l.logger.WithFields(fields),
+		name:         l.name,
+		logger:       l.logger.WithFields(fields),
+		traceEnabled: l.traceEnabled,
 	}
 }
 
+// WithContext return a logger with context.
+func (l *daprLogger) WithContext(ctx context.Context) Logger {
+	var id string
+
+	if l.traceEnabled {
+		id = trace.ID(ctx)
+	}
+
+	if id != "" {
+		return l.WithFields(map[string]any{
+			"id": id,
```

Comment:
Can we find a better name than `id`? How about `traceId`? Anything with the word "trace" in it